### PR TITLE
refactor(append_scylla_args): Cleanup all different usages of it

### PR DESCRIPTION
**Pull request by roydahan**
We had many different usages of scylla_args which may cause us to miss some issues in scylla.
Since we support thresholds for warnings and errors of reactor stalls, it's ok to have them all as low as 25 ms in test_default. This was the main usage of changeing append_scylla_args. Other usages like adding expermintal flags aren't relevant anymore. So, we left with 3 main "versions" of this parameter:
1. default - should be used in all of our tests.
2. Performance - Reducing stall detector to 5 ms and remove the "enable-sstable-key-validation" which I suspect can hurt performance.
3. Special - for example, set memory to 4GB.

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by roydahan** [comment]
This should be an FYI for everyone who test scylla, just to be aware that if there is a new failure, it may be due to enabling a missing flag and necessarily a new regression.

File: test-cases/scylla-operator/kubernetes-scylla-upgrade.yaml [hunk 1]
```diff
@@ -22,9 +22,10 @@ user_prefix: 'kubernetes-scylla-upgrade'
 #authenticator_user: 'cassandra'
 #authenticator_password: 'cassandra'
 
-# Does not work due to the lack of support, could be anabled after - https://github.com/scylladb/scylla-cluster-tests/issues/2747
-#append_scylla_args: '--blocked-reactor-notify-ms 500 --abort-on-lsa-bad-alloc 1 --abort-on-seastar-bad-alloc --abort-on-internal-error 1'
-
+# TODO: add '--abort-on-seastar-bad-alloc' arg to the 'append_scylla_args' option when
+#       following operator bug gets fixed: https://github.com/scylladb/scylla-operator/issues/991
+#       '--blocked-reactor-notify-ms 100' cannot be set, because it gets set by operator itself
+append_scylla_args: '--abort-on-lsa-bad-alloc 1 --abort-on-internal-error 1 --abort-on-ebadf 1 --enable-sstable-key-validation 1'
```
Comment (vponomaryov) [inline test-cases/scylla-operator/kubernetes-scylla-upgrade.yaml:28]: It is already defined in each of the K8S backends default config files.
So, if change this file then only by removing the red lines here.

**Review by soyacz — APPROVED** [review]

File: test-cases/scylla-operator/kubernetes-scylla-upgrade.yaml (hunk 1 above)
Comment (roydahan) [reply test-cases/scylla-operator/kubernetes-scylla-upgrade.yaml:28]: ok

**Comment by roydahan** [comment]
I'll wait for ack from @fruch @fgelcer @aleksbykov and @temichus

**Review by temichus — APPROVED** [review]

**Review by fruch — APPROVED** [review]
LGTM

**Comment by fruch** [comment]
@roydahan you still want this backported to v14 ?

**Comment by roydahan** [comment]
No

**Pull request merged**